controllers/componentsubscription_controller.go

```diff
@@ -66,58 +65,10 @@ func (r *ComponentSubscriptionReconciler) Reconcile(ctx context.Context, req ctr
 
 	// The replication controller doesn't need a shouldReconcile, because it should always reconcile,
 	// that is its purpose.
-	var patchHelper *patch.Helper
-	patchHelper, err = patch.NewHelper(obj, r.Client)
-	if err != nil {
-		conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.PatchFailedReason, err.Error())
-		return
-	}
+	patchHelper := patch.NewSerialPatcher(obj, r.Client)
 
 	// Always attempt to patch the object and status after each reconciliation.
 	defer func() {
```

Comment:
Is this even needed any longer? We could just patch when reconcile returns,

Reply:
True. Probably not needed anymore. :)

Reply:
Ah, but that would mean that we'll have to repeat the patch twice because we also want to patch when it fails and when it succeeds. And thus, this is cleaner. :)